service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/HearingBookingDetailsController.java

```diff
@@ -85,17 +105,9 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
         CaseDetails caseDetailsBefore = callbackRequest.getCaseDetailsBefore();
         CaseData caseDataBefore = mapper.convertValue(caseDetailsBefore.getData(), CaseData.class);
 
-        List<Element<HearingBooking>> hearingDetailsBefore = service.expandHearingBookingCollection(caseDataBefore);
-        List<Element<HearingBooking>> pastHearings = service.getPastHearings(hearingDetailsBefore);
-
-        List<Element<HearingBooking>> updatedHearings =
-            service.setHearingJudge(caseData.getHearingDetails(), caseData.getAllocatedJudge());
-
-        List<Element<HearingBooking>> combinedHearingDetails =
-            service.combineHearingDetails(updatedHearings, pastHearings);
-
-        caseDetails.getData().put(HEARING_DETAILS_KEY, combinedHearingDetails);
+        List<Element<HearingBooking>> hearingDetails = service.getHearingDetails(caseDetails, caseData, caseDataBefore);
```

Comment:
I agree that we shouldn't do too much in controller, but here we just put the whole controller logic into a service method, I don't think this is better

Reply:
Everything was in controller before, its much better for it to be in service. Ideally, i would want that some code from service delegated to other smaller units but as i was saying, we will not be able to do all that today. I was going to suggest we raise a tech debt and address it later. I can work on it today but don't want this to delay delivering business functionality. Happy to hear what others think!

Reply:
if there are no other big review comments to address then yes we can do it today.